model/controller.go

```diff
@@ -34,6 +36,9 @@ type Controller interface {
 	// for a service.
 	AppendInstanceHandler(f func(*ServiceInstance, Event)) error
 
+	// AppendIngressHandler notifies about changes to ingress resources
+	AppendIngressHandler(f func(*v1beta1.Ingress, Event)) error
+
```

Comment:
May be add a type to the Config struct to detect if its an ingress or not. This way, ingress becomes kube specific.